Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h

```diff
@@ -0,0 +1,453 @@
+       */
+      X_monotone_curve_2 operator()(const Point_2& p, const Point_2& q);
+
+      /*! Construct a full great circle from a normal to a plane.
+       * \param normal the normal to the plane containing the great circle.
+       * \pre the plane is not vertical.
+       */
+      X_monotone_curve_2 operator()(const Direction_3& normal);
+
+      /*! Construct a geodesic arc from two endpoints contained in a plane.
+       * \param[in] p the first endpoint.
+       * \param[in] q the second endpoint.
+       * \param[in] normal the normal to the plane containing the arc.
+       * \pre Both endpoints lie on the given plane.
+       */
+      X_monotone_curve_2 operator()(const Point_2& p, const Point_2& q,
+                                    const Direction_3& normal);
+
+      /// @} /* end of operations */
+    };
+
+    /*! Construction functor of geodesic arcs.
+     *
+     * \cgalModels `Assignable`
+     * \cgalModels `CopyConstructible`
+     * \cgalModels `AdaptableUnaryFunction`
+     * \cgalModels `AdaptableBinaryFunction`
+     * \cgalModels `AdaptableTernaryFunction`
+     */
+    class Construct_curve_2 {
+    public:
+      /// \name Types
+      /// @{
+      typedef Arr_geodesic_arc_on_sphere_traits_2<Kernel, X, Y>::Point_2 Point_2;
+      typedef Arr_geodesic_arc_on_sphere_traits_2<Kernel, X, Y>::Curve_2 result_type;
+      typedef Kernel::Direction_3 Direction_3;
+      typedef Direction_3 argument_type;
+      /// @}
+
+      /// \name Operations
+      /// @{
+
+      /*! Construct a full great circle from a normal to a plane.
+       * \param normal the normal to the plane containing the great circle.
+       */
+      X_monotone_curve_2 operator()(const Direction_3& normal);
+
+      /*! Construct the minor geodesic arc from two endpoints. The minor arc
+       * is the one with the smaller angle among the two geodesic arcs with
+       * the given endpoints.
+       * 1. Find out whether the arc is x-monotone.
+       * 2. If it is x-monotone,
+       *     1. Find out whether it is vertical, and
+       *     2. whether the target is larger than the source (directed right).
+       *
+       * An arc is vertical, iff
+       * 1. one of its endpoint direction pierces a pole, or
+       * 2. the projections of the endpoint directions onto the xy-plane coincide.
+       *
+       * \param[in] p the first endpoint.
+       * \param[in] q the second endpoint.
+       * \pre p and q must not coincide.
+       * \pre p and q cannot be antipodal.
+       */
+      Curve_2 operator()(const Point_2& p, const Point_2& q);
+
+      /*! Construct a geodesic arc from two endpoints contained
+       * in a plane.
+       * \param[in] p the first endpoint.
+       * \param[in] q the second endpoint.
+       * \param[in] normal the normal to the plane containing the arc.
+       * \pre Both endpoints lie on the given plane.
+       */
+      Curve_2 operator()(const Point_2& p, const Point_2& q,
```

Comment:
I still don't understand how to construct a curve describe the largest part of the circle between `p` and `q`.